### ISSUE TITLE
Support for ESW10-USA outlet.

### ISSUE THREAD

**Issue by linuxkd** [webdjoe/pyvesync#61]
Missing support for newer outlet version.

Version Installed:
pyvesync-1.2.1-py2.py3-none-any.whl

Output from logging into an account with the ESW10-USA plug active.
`>>> manager.update()`
`2020-11-03 13:21:21,918 - WARNING - Unknown device ESW10-USA`

**Comment by langholdt** [comment]
Leaving this comment here as well, I have one of these (actually 10 of them) and would love to help.  I can send a device or with instructions see if I can debug.  I'm an old coder for the 90's so this is truly a stretch for me.

Terry

**Comment by webdjoe** [comment]
If you want to capture packets I can add this- install the vesync APK on an Android Emulator (like Android Studio) using a generic android OS version. Then using frida as shown in Sections 4,5 in [Ultimate Guide](https://redhuntlabs.com/wp-content/uploads/2021/10/Ultimate-Guide-to-SSL-Pinning-Bypass-RedHunt-Labs-Attack-Surface-Management.pdf). After SSL pinning is disabled, capture packets with SSL proxy with charles or some other software that installs a root certificate.

Check out [this](https://infosecwriteups.com/hail-frida-the-universal-ssl-pinning-bypass-for-android-e9e1d733d29) for setting frida up. I never used genymotion, only Android Studio which requires a generic android OS to root the device.

If you are successful, please capture packets for all device functions and the device list screen.

**Comment by GOTAZ** [comment]
currently not available time wise to do the setup for packet capture. I can ship a unit to someone that already has a setup and knows what they are looking for.

**Comment by kevin-norlander** [comment]
Agreed. I don't have the time to setup a whole dev suite and capture packets right now. I'll ship one to anyone who has the time. Shoot, I'll send you a few that you can keep as a thank you for helping to fill this gap.

**Comment by webdjoe** [comment]
Create a new account with a throw away email and link that device. Email me at webdjoe at gmail with the details and I will capture packets tonight. I already have the env setup

*Edit be sure to unlink from your real account and you need an email address that you can receive emails it sends a verification code

**Comment by mjaverto** [comment]
@webdjoe did you get the above request? If not, I'm happy to create an account and link an ESW10-USA

**Comment by webdjoe** [comment]
Yes, this will be released with 3.0 in the next week or so

**Comment by kevin-norlander** [comment]
Just updated to 2025.03, hoping to be able to start controlling these plugs!!  Did the fix get integrated into this release?

**Comment by webdjoe** [comment]
I am refactoring the entire library, so it will be included in the next release. Time permitting, around a week.

**Comment by kevin-norlander** [comment]
Yeah - I got mixed up between repos and saw the HAAS update to 2025.03 which made me think about your 3. comment. Just a happy mistake... I'm excited for the next release!  :)

**Comment by marks-assistant** [comment]
The 2025.3 update broke all of the original outlets that used to be supported.  Restoring back to the 2025.2.5 fixes the issue.  This has been the case for every update after that release.  The original outlet is the one denoted as wifi-switch-1.3.  Note that I also have the outlets denoted as ESW03-USA and they work okay in the updated version.

**Comment by cdnninja** [comment]
@marks-assistant  could you link to the HA issue so I can take a look?

**Comment by kevin-norlander** [comment]
> The 2025.3 update broke all of the original outlets that used to be supported.  Restoring back to the 2025.2.5 fixes the issue.  This has been the case for every update after that release.  The original outlet is the one denoted as wifi-switch-1.3.  Note that I also have the outlets denoted as ESW03-USA and they work okay in the updated version.

I'm not having this same experience.  My counts are:

ESWL01 - 12
ESWL03 - 2
ESO15-TB - 4
ESWD16 - 10
EWS15-USA - 7
wifi-switch-1.3 - 7

All seem to be working well

HA Core 2025.4.3
Supervisor 2025.04.0
Operating System 15.2
Frontend 20250411.0

Just eagerly and patiently awaiting the device rewrite so I can start using my 13 ESW10-USA plugs with HA automations.

**Comment by marks-assistant** [comment]
I went back and dug into this and discovered that the non-working entities were recreated by the integration.  By deleting my original entities and adding the newly created ones everything is working again.  Not sure what caused it, but it does appear to be case sensitive.  Entities starting with lower case were recreated starting with upper case.  A strange one but the integration is now working with the update.  Thank you for the feedback,

**Comment by webdjoe** [comment]
The ESW10-USA is finally added, please test it out in the new branch: https://github.com/webdjoe/pyvesync/tree/dev-2.0. This is the new version of the library so please review the readme and documentation.

**Comment by GOTAZ** [comment]
Thanks for getting this fixed. Hopefully this doesn't sound to stupid, but
can someone dumb down the installation for me, as the installation is just
a one line command.   I assume I need to delete the old integration and
copy the folder to my custom components folder, but then what?

**Comment by webdjoe** [comment]
This is just the library, the home assistant integration still needs to be updated. You will have to clone the branch or use `pip install `  branch method and run a testing script like in the readme or documentation.

**Comment by cdnninja** [comment]
Reports that the ESW10-USA doesn't work for control.  I am waiting on debug logs: https://github.com/home-assistant/core/issues/135403#issuecomment-3536576974
